do not use the synaptics driver for devices advertising themselves as keyboards
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=524130

Index: xserver-xorg-input-synaptics/conf/11-x11-synaptics.fdi
===================================================================
--- xserver-xorg-input-synaptics.orig/conf/11-x11-synaptics.fdi	2009-11-21 13:15:39.064182104 +0900
+++ xserver-xorg-input-synaptics/conf/11-x11-synaptics.fdi	2009-11-21 13:16:23.404256090 +0900
@@ -2,27 +2,31 @@
 <deviceinfo version="0.2">
   <device>
     <match key="info.capabilities" contains="input.touchpad">
-        <merge key="input.x11_driver" type="string">synaptics</merge>
-        <!-- Arbitrary options can be passed to the driver using
-             the input.x11_options property since xorg-server-1.5. -->
-        <!-- EXAMPLES:
-	Maximum movement of the finger for detecting a tap
-	<merge key="input.x11_options.MaxTapMove" type="string">2000</merge>
-
-	Enable vertical scrolling when dragging along the right edge
-	<merge key="input.x11_options.VertEdgeScroll" type="string">true</merge>
-
-	Enable vertical scrolling when dragging with two fingers anywhere on the touchpad
-	<merge key="input.x11_options.VertTwoFingerScroll" type="string">true</merge>
-
-	Enable horizontal scrolling when dragging with two fingers anywhere on the touchpad
-	<merge key="input.x11_options.HorizTwoFingerScroll" type="string">true</merge>
-
-	If on, circular scrolling is used
-	<merge key="input.x11_options.CircularScrolling" type="string">true</merge>
-
-	For other possible options, check CONFIGURATION DETAILS in synaptics man page
-        -->
+      <!-- do not use the synaptics driver for devices advertising
+	   themselves as keyboards -->
+      <match key="info.capabilities" contains_not="input.keyboard">
+          <merge key="input.x11_driver" type="string">synaptics</merge>
+          <!-- Arbitrary options can be passed to the driver using
+               the input.x11_options property since xorg-server-1.5. -->
+          <!-- EXAMPLES:
+          Maximum movement of the finger for detecting a tap
+          <merge key="input.x11_options.MaxTapMove" type="string">2000</merge>
+
+          Enable vertical scrolling when dragging along the right edge
+          <merge key="input.x11_options.VertEdgeScroll" type="string">true</merge>
+
+          Enable vertical scrolling when dragging with two fingers anywhere on the touchpad
+          <merge key="input.x11_options.VertTwoFingerScroll" type="string">true</merge>
+
+          Enable horizontal scrolling when dragging with two fingers anywhere on the touchpad
+          <merge key="input.x11_options.HorizTwoFingerScroll" type="string">true</merge>
+
+          If on, circular scrolling is used
+          <merge key="input.x11_options.CircularScrolling" type="string">true</merge>
+
+          For other possible options, check CONFIGURATION DETAILS in synaptics man page
+          -->
+      </match>
     </match>
   </device>
 </deviceinfo>
